### ISSUE TITLE
Transport endpoint is not connected

### ISSUE THREAD

**Issue by raybus** [TurboGit/hubicfuse#98]
Hi!

I managed to mount my Hubic-cloudstorage to my Pi, with Jessie.
I also installed an FTP-server on it. 
When I try to download and upload from an FTP-client to my FTP-server to my Hubic-mount, it works for only 1 file. Then my FTP-server responds: Transport endpoint is not connected.
Als my mounting point from the command-line tells me Transport endpoint is not connected.
A solution I found for Fuse with this problem seems to be to add a parameter to the Fuse-mountcommand, -o reconnect. The Hubicfuse-help does not give me that option. 
My mounting-parameters are sudo hubicfuse /mnt/hubic -o noauto_cache,sync_read,allow_other
Does anybody has an idea how to fix this within hubicfuse?
I already know that unmounting and mounting again will fix it, but that's only temporarily.

**Comment by raybus** [comment]
Extra info... 
When I copy 3 files from my Pi to the mount, just from commandline by cp... I get the following erros:
cp: failed to close ‘/mnt/hubic/default/test/favicon.png’: Software caused connection abort
cp: cannot stat ‘/mnt/hubic/default/test/logo.jpg’: Transport endpoint is not connected
cp: cannot stat ‘/mnt/hubic/default/test/logo.png’: Transport endpoint is not connected

It seems he cannot close the first file so he stops.

**Comment by dluyten** [comment]
Exactly the same issue and no idea how to solve it

**Comment by dan-cristian** [comment]
Can you try to enable debugging and post the debug output?

**Comment by oursgris** [comment]
hello
i have the same problem

tryied to add the debug_level option but there was no difference or log ..

pi@raspberrypi / $ sudo hubicfuse /mnt/h -o noauto_cache,sync_read,allow_other,debug_level=1
pi@raspberrypi / $ sudo dd if=/dev/zero of=/mnt/hubic/default/te bs=512k count=200
dd: impossible d'ouvrir « /mnt/hubic/default/te »: Noeud final de transport n'est pas connecté

please note that after that I can go several minutes in the /mnt/h directory

for ex : 
pi@raspberrypi /mnt/h/default $ ls
test  Backup
pi@raspberrypi /mnt/h/default $ mkdir a
pi@raspberrypi /mnt/h/default $ ls
a     test    Backup

but if i copy something no way
pi@raspberrypi /mnt/h/default/a $ cp /home/pi/1f.rb ./
cp: impossible de fermer « ./1f.rb »: Le logiciel a provoqué l'abandon de la connexion

**Comment by dan-cristian** [comment]
You have to run hubicfuse with -d or -f parameter to enable debug output.

**Comment by oursgris** [comment]
Ok

I removed the different id which seemed personal

pi@raspberrypi /mnt/h/default/a $ sudo hubicfuse /mnt/j2 -o  noauto_cache,sync_read,allow_other,debug_level=1 -d &
debug_level = 1
get_extended_metadata = 0
curl_progress_state = 0
enable_chmod = 0
enable_chown = 0
==DBG 0 [2015-12-15 10:39:17.]:6908==Authenticating... (client_id = 'api_hubic**********')
==DBG 0 [2015-12-15 10:39:18.]:6908==HUBIC TOKEN_URL result: '{"expires_in":21600,"access_token":"**********","token_type":"Bearer"}'

==DBG 0 [2015-12-15 10:39:18.]:6908==HUBIC Access token: **********

==DBG 0 [2015-12-15 10:39:18.]:6908==HUBIC Token type  : Bearer

==DBG 0 [2015-12-15 10:39:18.]:6908==HUBIC Expire in   : 21600

==DBG 1 [2015-12-15 10:39:18.]:6908==add_header(Authorization:Bearer **********)
==DBG 0 [2015-12-15 10:39:18.]:6908==CRED_URL result: '{"token":"**********","endpoint":"http                                                                                                                          s://lb1.hubic.ovh.net/v1/AUTH_**********","expires":"2015-12-16T08:35:34+01:00"}'

FUSE library version: 2.9.3
nullpath_ok: 0
nopath: 0
utime_omit_ok: 0
unique: 1, opcode: INIT (26), nodeid: 0, insize: 56, pid: 0
INIT: 7.23
flags=0x0003f7fb
max_readahead=0x00020000
   INIT: 7.19
   flags=0x00000010
   max_readahead=0x00020000
   max_write=0x00020000
   max_background=0
   congestion_threshold=0
   unique: 1, success, outsize: 40

pi@raspberrypi /mnt $ cd j2

unique: 3, opcode: GETATTR (3), nodeid: 1, insize: 56, pid: 4517
getattr /
==DBG 0 [2015-12-15 10:41:40.]:6912==cfs_getattr(/)
==DBG 0 [2015-12-15 10:41:40.]:6912==exit 0: cfs_getattr(/)
   unique: 3, success, outsize: 120
unique: 4, opcode: ACCESS (34), nodeid: 1, insize: 48, pid: 4517
   unique: 4, error: -38 (Function not implemented), outsize: 16

pi@raspberrypi /mnt/j2 $ ls
unique: 5, opcode: OPENDIR (27), nodeid: 1, insize: 48, pid: 7038
   unique: 5, success, outsize: 32
unique: 6, opcode: READDIR (28), nodeid: 1, insize: 80, pid: 7038
readdir[0] from 0
==DBG 0 [2015-12-15 10:42:23.]:6911==cfs_readdir(/)
==DBG 1 [2015-12-15 10:42:23.]:6911==caching_list_directory(/)
==DBG 1 [2015-12-15 10:42:23.]:6911==cloudfs_list_directory()
==DBG 1 [2015-12-15 10:42:23.]:6911==send_request_size(GET) (/?format=xml)
==DBG 1 [2015-12-15 10:42:23.]:6911==add_header(X-Auth-Token:73c8e338403f4f7499889aaffc4c36c7)
==DBG 1 [2015-12-15 10:42:23.]:6911==check_path_info()
==DBG 1 [2015-12-15 10:42:23.]:6911==check_caching_list_directory()
==DBG 1 [2015-12-15 10:42:23.]:6911==exit 1: check_caching_list_directory() [CACHE-DIR-MISS]
==DBG 1 [2015-12-15 10:42:23.]:6911==exit 0: check_path_info() [CACHE-MISS]
==DBG 1 [2015-12-15 10:42:23.]:6911==send_request_size: GET XML (/?format=xml)
==DBG 1 [2015-12-15 10:42:23.]:6911==status: send_request_size(/?format=xml) started HTTP REQ:https://lb1.hubic.ovh.net/v1/AUTH_******************/?format=xml
==DBG 1 [2015-12-15 10:42:24.]:6911==status: send_request_size(/?format=xml) completed HTTP REQ:https://lb1.hubic.ovh.net/v1/AUTH_*******************a/?format=xml total_time=0.3 seconds
==DBG 0 [2015-12-15 10:42:24.]:6911==exit 0: send_request_size(/?format=xml) speed=0.3 sec (GET) [HTTP OK]
==DBG 1 [2015-12-15 10:42:24.]:6911==ls: lecture du répertoire .: Le logiciel a provoqué l'abandon de la connexion

pi@raspberrypi /mnt/j2 $ ls
ls: impossible d'ouvrir le répertoire .: Noeud final de transport n'est pas connecté
[1]+  Termine 139             sudo hubicfuse /mnt/j2 -o noauto_cache,sync_read,allow_other,debug_level=1 -d  (wd : /mnt/h/default/a)
(maintenant, wd : /mnt/j2)
pi@raspberrypi /mnt/j2 $ ls
ls: impossible d'ouvrir le répertoire .: Noeud final de transport n'est pas connecté

the mount was killed when I type ls the second type

**Comment by raybus** [comment]
Hi Guys!
To be honest I am not that Linux-guru, so thanks for all the technical contributions! I will monitor this issue from the sideway but not less interested! :)

**Comment by dan-cristian** [comment]
from the debug output is not obvious what the issue is as no error is thrown. need more debug info to identify the issue.

can you try running hubicfuse under a debugger: gdb, as shown in the script below. change it as per your linux configs. ideally build hubicfuse with make debug.

https://github.com/dan-cristian/hubicfuse/blob/master/debug.sh

start the process with run after debugger initialise and once hubicfuse exits type: "back" to show the stacktrace.

**Comment by oursgris** [comment]
althought i used freebsd a lot in the past times, i'm sorry i tryied but i didn't manage it

I made the following :
git clone https://github.com/TurboGit/hubicfuse.git
cd hubicfuse
./configure
./debug.sh

root@raspberrypi:/opt/hubicfuse/hubicfuse# ./debug.sh
hubicfuse: aucun processus trouvé
fatal: Couldn't find remote ref utime
gcc -g -O2 -I/usr/include/libxml2   -D_FILE_OFFSET_BITS=64 -I/usr/include/fuse   -I/usr/include/json-c  -g -o hubicfuse cloudfsapi.c cloudfuse.c commonfs.c -lxml2  -lcurl  -lfuse -pthread  -lssl -lcrypto  -ljson-c  -lmagic
umount: /mnt/hubic: not mounted
/usr/bin/install -c hubicfuse /usr/local/bin/hubicfuse
GNU gdb (Raspbian 7.7.1+dfsg-5) 7.7.1
Copyright (C) 2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "arm-linux-gnueabihf".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from hubicfuse...done.
(gdb)

and it stay of that line ..
I don't know what to do

if you are interested, I can give you a teamviewer access to my desktop
(I can't give you my ssh)
note : I use a raspberry pi

**Comment by dan-cristian** [comment]
type "run" and press enter after you are prompted with (gdb) and hubicfuse
should start.

but change the debug.sh script to match your local paths.

**Comment by oursgris** [comment]
(gdb) run
Starting program: /opt/hubicfuse/hubicfuse/hubicfuse /mnt/hubic -o noauto_cache,sync_read,allow_other -f
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".
Cannot access memory at address 0x0

Program received signal SIGILL, Illegal instruction.
0x76c38d08 in ?? () from /usr/lib/arm-linux-gnueabihf/libcrypto.so.1.0.0
(gdb) run
The program being debugged has been started already.
Start it from the beginning? (y or n) y
Starting program: /opt/hubicfuse/hubicfuse/hubicfuse /mnt/hubic -o noauto_cache,sync_read,allow_other -f
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/arm-linux-gnueabihf/libthread_db.so.1".
Cannot access memory at address 0x0

Program received signal SIGILL, Illegal instruction.
0x76c38d08 in ?? () from /usr/lib/arm-linux-gnueabihf/libcrypto.so.1.0.0

**Comment by oursgris** [comment]
the /mnt/hubic is fine
the directory exists

**Comment by TurboGit** [comment]
Looks like a very bad issue:

> Program received signal SIGILL, Illegal instruction.
> 0x76c38d08 in ?? () from /usr/lib/arm-linux-gnueabihf/libcrypto.so.1.0.0

Maybe a wrong compilation for this unit or an alignment issue. I'm no expert in arm-linux so I cannot really help on this one.

Can you just after the "run" enter "bt" (backtrace).

No need to run multiple time. When a SIGILL is encountered there is no need to go further. The backtrace will give us the point of call where all goes wrong.

**Comment by oursgris** [comment]
Program received signal SIGILL, Illegal instruction.
0x76c38d08 in ?? () from /usr/lib/arm-linux-gnueabihf/libcrypto.so.1.0.0
(gdb) bt
#0  0x76c38d08 in ?? () from /usr/lib/arm-linux-gnueabihf/libcrypto.so.1.0.0
#1  0x76c353f4 in OPENSSL_cpuid_setup () from /usr/lib/arm-linux-gnueabihf/libcrypto.so.1.0.0
#2  0x76fdf058 in call_init (l=<optimized out>, argc=5, argv=0x7efff774, env=0x7efff78c) at dl-init.c:78
#3  0x76fdf134 in _dl_init (main_map=0x76fff958, argc=5, argv=0x7efff774, env=0x7efff78c) at dl-init.c:126
#4  0x76fcfda4 in _dl_start_user () from /lib/ld-linux-armhf.so.3
Backtrace stopped: previous frame identical to this frame (corrupt stack?)

**Comment by oursgris** [comment]
I imagine we don't use the same libcrypto ..
I'm on arm with raspbian

is it the libcrypt++-dev package ?

root@raspberrypi:/opt/hubicfuse/hubicfuse# apt-cache show libcrypto++-dev
Package: libcrypto++-dev
Source: libcrypto++
Version: 5.6.1-6+deb8u1
Architecture: armhf

**Comment by oursgris** [comment]
i saw someone with exactly the same issue on raspberry with another software
https://github.com/jpmens/mqttcollect/issues/5
can you look at it ?

**Comment by netomx** [comment]
Is there any way to make a keepalive?

**Comment by netomx** [comment]
[code]unique: 9806, opcode: WRITE (16), nodeid: 4, insize: 61057, pid: 610
write[140701450898496] 60977 bytes to 641204224 flags: 0x8001
   write[140701450898496] 60977 bytes to 641204224
   unique: 9806, success, outsize: 24
unique: 9807, opcode: SETATTR (4), nodeid: 4, insize: 128, pid: 610
utimens /default/One-Punch Man - 01.mkv 1450307085.656615735 1450306894.326101387
==DBG 0 [2015-12-16 18:10:33.]:612==cfs_utimens(/default/One-Punch Man - 01.mkv)
==DBG 0 [2015-12-16 18:10:33.]:612==exit 1: cfs_utimens(/default/One-Punch Man - 01.mkv)
getattr /default/One-Punch Man - 01.mkv
==DBG 0 [2015-12-16 18:10:33.]:612==cfs_getattr(/default/One-Punch Man - 01.mkv)
==DBG 0 [2015-12-16 18:10:33.]:612==send_request_size: not found error for (GET)(default/One-Punch%20Man%20-%2001.mkv), ignored [HTTP 404].
==DBG 0 [2015-12-16 18:10:33.]:612==exit 2: cfs_getattr(/default/One-Punch Man - 01.mkv)
   unique: 9807, success, outsize: 120
unique: 9808, opcode: SETXATTR (21), nodeid: 4, insize: 100, pid: 610
   unique: 9808, error: -38 (Function not implemented), outsize: 16
unique: 9809, opcode: SETATTR (4), nodeid: 4, insize: 128, pid: 610
==DBG 0 [2015-12-16 18:10:33.]:593==cfs_chmod(/default/One-Punch Man - 01.mkv,33188)
==DBG 0 [2015-12-16 18:10:33.]:593==cfs_chmod(/default/One-Punch Man - 01.mkv): change mode from 33152 to 33188
==DBG 0 [2015-12-16 18:10:34.]:593==send_request_size: not found error for (PUT)(default/One-Punch%20Man%20-%2001.mkv), ignored [HTTP 404].
getattr /default/One-Punch Man - 01.mkv
==DBG 0 [2015-12-16 18:10:34.]:593==cfs_getattr(/default/One-Punch Man - 01.mkv)
==DBG 0 [2015-12-16 18:10:34.]:593==exit 2: cfs_getattr(/default/One-Punch Man - 01.mkv)
   unique: 9809, success, outsize: 120
unique: 9810, opcode: FLUSH (25), nodeid: 4, insize: 64, pid: 610
flush[140701450898496]
==DBG 0 [2015-12-16 18:10:34.]:592==cfs_flush(/default/One-Punch Man - 01.mkv)
==DBG 0 [2015-12-16 18:10:34.]:592==file_md5: 0 file buffer size, using default
==DBG 0 [2015-12-16 18:10:35.]:592==cfs_flush(/default/One-Punch Man - 01.mkv): perform full upload as content changed (or no file found in cache)
[/code]

**Comment by dan-cristian** [comment]
I do not understand the question, can you clarify?

**Comment by netomx** [comment]
Sorry about the keepalive, I have seen now that the problem is on the code, debuggint it, HubiC sends 404 errors and your module doesn't return any data, and it crashes.

**Comment by dan-cristian** [comment]
404 errors are thrown usually when there is no file found in the cloud storage, if you don't provide any debug output is not possible to tell if is a hubicfuse issue or not.

**Comment by robot3498712** [comment]
Hi, same issue as mentioned here on a raspi2 (same specs oursgris posted earlier).

Debug log:
https://cryptobin.org/15t777k8
password: hubicfuse

root@aiko:~# hubicfuse /mnt/hubic -d -o noauto_cache,sync_read,allow_other >>/var/log/hubic_debug.log 2>&1
Segmentation fault

root@aiko:/tmp# cp *.test /mnt/hubic/tmp/
cp: failed to close ‘/mnt/hubic/tmp/file1.test’: Software caused connection abort
cp: cannot stat ‘/mnt/hubic/tmp/file2.test’: Transport endpoint is not connected

If someone can look into this and provide a fix it'd be much appreciated!

**Comment by Fraifrai** [comment]
I have the same issue using Gentoo GNU/Linux : no problem  while reading files but after the first file writing the connexion is closed (but the file is well uploaded)

**Comment by Fraifrai** [comment]
Funny: using rsync it creates directories but still hang after the copy of the first uploaded file

**Comment by Fraifrai** [comment]
At least I got a log !!! https://framabin.org/?3e2f474aa77f8101#j7axmkRuz0oBuOenK+AGAmFF2Tjr6XtnbhiclCXJn0E=

**Comment by snyfear** [comment]
Same issue as Fraifrai but on Xubuntu 15.10 : "no problem while reading files but after the first file writing the connexion is closed (but the file is well uploaded)"

**Comment by dluyten** [comment]
Exactly the same for me

**Comment by krnflake** [comment]
Looks like a newer kernel like 4.3.x is causing the problems, works fine on 4.2.x.

**Comment by TurboGit** [comment]
No, I'm using 4.3.x on Debian/sid since some time without trouble.

**Comment by krnflake** [comment]
@TurboGit could you please tell me which kernel version you're currently using? Hubicfuse worked fine for me using some 4.2.x kernel, but since I've switched to 4.3.3 i'm having the same problems as described above.

**Comment by oursgris** [comment]
@krnflake are you on raspberry or any other arm platform ?

**Comment by krnflake** [comment]
@oursgris no, I'm on x64, I was refering to @Fraifrai 's issue.

**Comment by TurboGit** [comment]
@krnflake I'm using exactly "SMP Debian 4.3.3-2 (2015-12-17) x86_64 GNU/Linux"

Note that you may need to recompile hubicfuse when the kernel is upgraded. Did you do that?

**Comment by Fraifrai** [comment]
It works in a Debian8 VBox machine with kernel 3.16,fuse 2.9.3-15+deb8u1,

it does'nt work in my native genoo kernel 4.1.12, fuse 2.9.4

**Comment by TurboGit** [comment]
But again there is something strange in this report. I'm using hubicfuse in a Debian/testing and Debian//sid since years. Two rolling releases updated almost daily and I never had this problem. Or each time I had an issue I just needed to recompile hubicfuse with the new kernel.

**Comment by TurboGit** [comment]
Just a question, is this issue only happening with rsync?

**Comment by TurboGit** [comment]
And even though, rsync works for me:

> $ rsync -a --progress *.png /mnt/hubic/default/tmppng/
> sending incremental file list
> created directory /mnt/hubic/default/tmppng
> Capture d'écran de 2015-11-12 18-47-24.png
        640,849 100%   82.84MB/s    0:00:00 (xfr#1, to-chk=9/10)
Capture d'écran de 2015-11-18 13-13-19.png
        903,556 100%   94.34kB/s    0:00:09 (xfr#2, to-chk=8/10)
Capture d'écran de 2015-12-13 16-59-08.png
      2,012,105 100%  166.38kB/s    0:00:11 (xfr#3, to-chk=7/10)
Capture d'écran de 2015-12-26 09-46-45.png
        234,097 100%   10.52kB/s    0:00:21 (xfr#4, to-chk=6/10)
Capture d'écran de 2015-12-26 09-49-31.png
        144,536 100%   35.53kB/s    0:00:03 (xfr#5, to-chk=5/10)
ada2012png-black.png
         17,456 100%  266.36kB/s    0:00:00 (xfr#6, to-chk=4/10)
by-nc-sa.eu.png
         11,287 100%  141.31kB/s    0:00:00 (xfr#7, to-chk=3/10)
favicon.png
            417 100%    4.24kB/s    0:00:00 (xfr#8, to-chk=2/10)
pm-screen-shot1.png
        370,657 100%   31.34kB/s    0:00:11 (xfr#9, to-chk=1/10)
pm-screen-shot2.png
         19,133 100%  316.69kB/s    0:00:00 (xfr#10, to-chk=0/10)

Second try:

> $ rsync -a --progress *.png /mnt/hubic/default/tmppng/
sending incremental file list

Let's see what is uploaded:

> $ ls -al /mnt/hubic/default/tmppng/
total 4276
drwxr-xr-x 2 obry obry       0 Jan  6 10:43 .
drwxr-xr-x 2 obry obry       0 Jan  6 10:43 ..
-rw-rw-rw- 1 obry obry   17456 Aug 30  2012 ada2012png-black.png
-rw-rw-rw- 1 obry obry   11287 Nov 28  2014 by-nc-sa.eu.png
-rw-rw-rw- 1 obry obry  640849 Nov 12 18:47 Capture d'écran de 2015-11-12 18-47-24.png
-rw-rw-rw- 1 obry obry  903556 Nov 18 13:13 Capture d'écran de 2015-11-18 13-13-19.png
-rw-rw-rw- 1 obry obry 2012105 Dec 13 16:59 Capture d'écran de 2015-12-13 16-59-08.png
-rw-rw-rw- 1 obry obry  234097 Dec 26 09:46 Capture d'écran de 2015-12-26 09-46-45.png
-rw-rw-rw- 1 obry obry  144536 Dec 26 09:49 Capture d'écran de 2015-12-26 09-49-31.png
-rw-rw-rw- 1 obry obry     417 Mar  5  2011 favicon.png
-rw-rw-rw- 1 obry obry  370657 Oct  7 09:32 pm-screen-shot1.png
-rw-rw-rw- 1 obry obry   19133 Oct  7 09:33 pm-screen-shot2.png

And all looks good...

**Comment by Fraifrai** [comment]
On my gentoo this issue occurs with both cp and rsync

On Debian : no problem

**Comment by TurboGit** [comment]
What is fuse version on both?

**Comment by krnflake** [comment]
There's a problem with GCC's Stack Smashing, which should terminate the program in case of buffer overflows. It gets triggered but I can't find a buffer overflow :cry:. I've recompiled hubicfuse and disabled Stack Smashing using this flag:

```
-fno-stack-protector
```

and now it's working perfectly, but I wouldn't recommend turning off this security feature.

**Comment by dan-cristian** [comment]
There might be some memory leaks as I have run hubicfuse under valgrind and got plenty of warnings.

I’m currently working to refactor hubicfuse and introduce progressive operations, will try to look also at memory leaks but for sure will take time.

 

Did someone configured a valgrind supression file for hubicfuse to filter out the false warnings (e.g. libcurl produces a lot of them)?

**Comment by Fraifrai** [comment]
@TurboGit : about fuse versions https://github.com/TurboGit/hubicfuse/issues/98#issuecomment-169273329

**Comment by TurboGit** [comment]
Sorry missed that :) So we are using the same fuse version... Even more confused now :)

**Comment by Fraifrai** [comment]
@TurboGit I notice again that it works perfectly under my debian and fails only with gentoo.

**Comment by TurboGit** [comment]
Yes, sure but then there is some difference that are to be found. The fuse version is the same, fine. The kernel used on gentoo is not the one I use today on debian but I have used it without trouble. So what? Another dependency? Maybe you can check the version on Debian vs gentoo of dependencies listed in the REAME? Maybe ssl?

**Comment by raybus** [comment]
And where does my Raspberry Pi come into this? It has the DEBIAN Jessie image on it...

**Comment by Fraifrai** [comment]
@TurboGit I'm not home but I remember that the stack suggests that it's a problem of threaded ssl processes

**Comment by TurboGit** [comment]
Indeed, maybe the issue! If OpenSSL is not compiled with thead safety it may explain the issue.

With the following program (main.c):

#include <stdio.h>

#define OPENSSL_THREAD_DEFINES
#include <openssl/opensslconf.h>

void main (void)
{
#if defined(OPENSSL_THREADS)
  printf("thread support enabled\n");
#else
  printf("no thread support\n");
#endif
}

What the output for:

$ gcc main.c && ./a.out

**Comment by robot3498712** [comment]
#include <openssl/ssl.h>
#define OPENSSL_THREAD_DEFINES
#include <stdio.h>
(..)

gcc ssl_test.c && ./a.out
thread support enabled

I also tried -fno-stack-protector ; apparently no such option on Linux 4.1.13-v7+ armv7l

**Comment by Fraifrai** [comment]
fraifrai@fraifrai ~/tmp $ gcc main.c 
main.c:1:27: erreur fatale:  openssl/ssl.h  : Aucun fichier ou dossier de ce type
 #include < openssl/ssl.h >

**Comment by Fraifrai** [comment]
I got thread support enabled

**Comment by TurboGit** [comment]
ok, so that's not the issue :(

**Comment by Fraifrai** [comment]
??? I believed that the issue what caused by multi-threaded mode

**Comment by TurboGit** [comment]
No, the possible issue was that the OpenSSL library was compiled *without* threaded support. But if it has threaded mode all should be fine.

**Comment by lupoair** [comment]
I'm getting the same error: *** stack smashing detected ***: hubicfuse terminated
And I was wondering how do I get OpenSSL with threaded support?

**Comment by maxegger** [comment]
Hello, 
I had the same problem under CentOS.
Try compile hubicfuse with O1 option (oryginally in Makefile is O2):

CFLAGS="-O1" PKG_CONFIG_PATH=/lib64/pkgconfig ./configure
make & make install

... and problem solved

**Comment by lupoair** [comment]
#107

**Comment by dan-cristian** [comment]
just submitted a fix merge request for #107 which might solve this problem as well

**Comment by Fraifrai** [comment]
Thanks !! it solved the issue. It can be closed !!!

**Comment by Nihim** [comment]
I do hope I am not reopening a ticket without need.

After several days (weeks?) of hubicfuse working fine, I got the above error on a debian 8 vps.
In my case it's like hubicfuse detached? the folder / files dissappeared though it still appeared on mount. I just connected with sftp and it was empty, logged in and indeed the dirs were nowhere to be found.

umounting, re mounting instantly fixed it.

I am entry-level in linux but a quick look at syslog doesn't show anything.
